src/rgw/rgw_aio.cc

```diff
@@ -0,0 +1,69 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2018 Red Hat, Inc.
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation. See file COPYING.
+ *
+ */
+
+#include "include/rados/librados.hpp"
+
+#include "rgw_aio.h"
+
+using std::unique_ptr;
+
+namespace rgw {
+// Annoyingly we have to allocate this little chunk of stuff on the
```

Comment:
still?  I was hoping we were removing the need to heap allocate ops or their completions